synapse/handlers/federation_event.py

```diff
@@ -416,6 +416,18 @@ async def backfill(
         events = await self._federation_client.backfill(
```

Comment:
The conversation for topological sorting when we receive backfill events continues in https://github.com/matrix-org/synapse/pull/11114#discussion_r739012329

---

For chunking and online topological sorting, we [discussed this in an out of band call](https://docs.google.com/document/d/1KCEmpnGr4J-I8EeaVQ8QJZKBDu53ViI7V62y5BzfXr0/edit#heading=h.1i5kq1pqrh5h):

> Plan forward:
>
>  1. Go forward with [this PR] so it fixes the scrollback case in federation
>  1. For Gitter and the static archives, we can have a script manually scrollback across all of the rooms in the archive server before anyone else or Google spider crawls in some weird way. This way it will lock the sort in place for all of the historical messages
>  1. Online topological ordering which will apply retroactively to fix any inconsistencies introduced by people permalinking
> 
> [*-- meeting summary notes 2021-10-28*](https://docs.google.com/document/d/1KCEmpnGr4J-I8EeaVQ8QJZKBDu53ViI7V62y5BzfXr0/edit#heading=h.1i5kq1pqrh5h)